src/sql-parser/src/parser.rs

```diff
@@ -2532,9 +2532,12 @@ impl Parser {
 
     fn parse_show(&mut self) -> Result<Statement, ParserError> {
         if self.parse_keyword("DATABASES") {
-            return Ok(Statement::ShowDatabases(ShowDatabasesStatement {
-                filter: self.parse_show_statement_filter()?,
-            }));
+            let filter = if self.parse_keyword("WHERE") || self.parse_keyword("LIKE") {
```

Comment:
This doesn't seem like it handles `LIKE` right? Like, probably you need to leave the AST as is and unpack the `ShowStatementFilter` into the appropriate `WHERE` clause in the sql crate? But this seems well tested, so as long as those tests are passing I'm good with it!